src/SearchResults.js

```diff
@@ -0,0 +1,83 @@
+import React from "react";
+import moment from "moment";
+
+class SearchResults extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      selectedRows: []
+    };
+  }
+
+  onClickHandler = clickedId => {
+    let selectedRows = [].concat(this.state.selectedRows);
+    const selectedIndex = selectedRows.findIndex(id => {
+      return id === clickedId;
+    });
+
+    /**
+     *  if row already selected
+     *  remove from array of selected rows
+     */
+
+    if (selectedIndex !== -1) {
+      selectedRows.splice(selectedIndex, 1);
+    } else {
+      // if not already selected, add to array
+      selectedRows.push(clickedId);
+    }
+
+    this.setState({ selectedRows });
+  };
+
+  render() {
+    return (
+      <table className="table table-dark">
+        <thead>
+          <tr>
+            <th scope="col">id</th>
+            <th scope="col">title</th>
+            <th scope="col">first name</th>
+            <th scope="col">surname</th>
+            <th scope="col">email</th>
+            <th scope="col">room id</th>
+            <th scope="col">check in date</th>
+            <th scope="col">check out date</th>
+            <th scope="col">Stay</th>
+          </tr>
+        </thead>
+        <tbody>
+          {this.props.results.map(result => {
+            const isSelected = this.state.selectedRows.find(id => {
+              return id === result.id;
```

Comment:
Really great job! Well done figuring this out - it is quite a tricky problem :)

However there is actually a much simpler approach to this problem. Do you remember in class we had the example with multiple Counter components? Each counter remembered their own count.

We can do a similar thing here - if we extracted a table row component (one for each result), then they could each remember their own state. In fact they would only need to remember if they were selected or not (which sounds quite like a boolean to me 😉). How would we go about implementing that?